packages/vite/src/node/plugins/worker.ts

```diff
@@ -287,25 +287,35 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
           : 'classic'
         : 'module'
       const workerOptions = workerType === 'classic' ? '' : ',{type: "module"}'
+
       if (isBuild) {
         getDepsOptimizer(config, ssr)?.registerWorkersSource(id)
         if (query.inline != null) {
           const chunk = await bundleWorkerEntry(config, id, query)
-          // inline as blob data url
-          return {
-            code: `const encodedJs = "${Buffer.from(chunk.code).toString(
-              'base64',
-            )}";
-            const blob = typeof window !== "undefined" && window.Blob && new Blob([atob(encodedJs)], { type: "text/javascript;charset=utf-8" });
-            export default function WorkerWrapper() {
-              const objURL = blob && (window.URL || window.webkitURL).createObjectURL(blob);
-              try {
-                return objURL ? new ${workerConstructor}(objURL) : new ${workerConstructor}("data:application/javascript;base64," + encodedJs${workerOptions});
-              } finally {
-                objURL && (window.URL || window.webkitURL).revokeObjectURL(objURL);
-              }
-            }`,
+          const encodedJs = `const encodedJs = "${Buffer.from(
+            chunk.code,
+          ).toString('base64')}";`
+
+          const blobCode = `${encodedJs}
+          const blob = typeof window !== "undefined" && window.Blob && new Blob([atob(encodedJs)], { type: "text/javascript;charset=utf-8" });
+          export default function WorkerWrapper() {
+            const objURL = blob && (window.URL || window.webkitURL).createObjectURL(blob);
+            try {
+              return objURL ? new ${workerConstructor}(objURL) : new ${workerConstructor}("data:application/javascript;base64," + encodedJs${workerOptions});
+            } finally {
+              objURL && (window.URL || window.webkitURL).revokeObjectURL(objURL);
+            }
```

Comment:
```js
try {
  try {
    if (!objURL) throw ''
    return new ${workerConstructor}(objURL)
  } catch {
    return new ${workerConstructor}("data:application/javascript;base64," + encodedJs${workerOptions});
  }
} finally {
  objURL && (window.URL || window.webkitURL).revokeObjectURL(objURL);
}
```
I think we need a try-catch like this to support #12002. I haven't tested whether an error is thrown in this case though.

The current code doesn't fallback to data-uri worker when the blob worker throwed an error.

Reply:
I still could not find reproduce it but I found a similar question https://stackoverflow.com/questions/11420846/whats-wrong-with-my-code-to-record-audio-in-html5. It points to `createObjectURL(stream)` throwing the error so I put that statement into the try-block. 
Also, I think using one try-catch block is enough here, I made a tweak based on your suggestions. Take a look. 😀